Xamarin.Forms.Core/SwipeItem.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.ComponentModel;
+using System.Windows.Input;
+
+namespace Xamarin.Forms
+{
+	public interface ISwipeItem
+	{
+		ICommand Command { get; set; }
+		object CommandParameter { get; set; }
+
+		event EventHandler<SwipeItemInvokedEventArgs> Invoked;
+	}
+
+	public class SwipeItem : ContextItem, ISwipeItem
+	{
+		public static readonly BindableProperty BackgroundColorProperty = BindableProperty.Create(nameof(BackgroundColor), typeof(Color), typeof(SwipeItem), Color.Default);
+
+		public Color BackgroundColor
+		{
+			get { return (Color)GetValue(BackgroundColorProperty); }
+			set { SetValue(BackgroundColorProperty, value); }
+		}
+
+		public event EventHandler<SwipeItemInvokedEventArgs> Invoked;
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public void OnInvoked() => Invoked?.Invoke(this, new SwipeItemInvokedEventArgs(this));
+	}
+
+	public class CustomSwipeItem : ContentView, ISwipeItem
```

Comment:
Not sure we need to expose this, but if we do, that's not a good name

Reply:
Mnn, do you have any suggestions for the name? I will think about better options.